commands/triggers.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2018 The Doctl Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/digitalocean/doctl"
+	"github.com/digitalocean/doctl/commands/displayers"
+	"github.com/spf13/cobra"
+)
+
+// Triggers generates the serverless 'triggers' subtree for addition to the doctl command
+func Triggers() *Command {
+	cmd := &Command{
+		Command: &cobra.Command{
+			Use:   "triggers",
+			Short: "Manage triggers associated with your functions",
+			Long: `When Functions are deployed by ` + "`" + `doctl serverless deploy` + "`" + `, they may have associated triggers.
+The subcommands of ` + "`" + `doctl serverless triggers` + "`" + ` are used to list and inspect
+triggers.  Each trigger has an event source type, and invokes its associated function
+when events from that source type occur.  Currently, only the ` + "`" + `scheduler` + "`" + ` event source type is supported.`,
+			Aliases: []string{"trig"},
+			Hidden:  true, // trigger support uses APIs that are not yet universally available
+		},
+	}
+	list := CmdBuilder(cmd, RunTriggersList, "list", "Lists your triggers",
+		`Use `+"`"+`doctl serverless triggers list`+"`"+` to list your triggers.`,
+		Writer, displayerType(&displayers.Triggers{}))
+	AddStringFlag(list, "function", "f", "", "list only triggers for the chosen function")
+
+	CmdBuilder(cmd, RunTriggersGet, "get <triggerName>", "Get the details for a trigger",
+		`Use `+"`"+`doctl serverless triggers get <triggerName>`+"`"+` for details about <triggerName>.`,
+		Writer)
+
+	return cmd
+}
+
+// RunTriggersList provides the logic for 'doctl sls trig list'
+func RunTriggersList(c *CmdConfig) error {
+	if len(c.Args) > 0 {
+		return doctl.NewTooManyArgsErr(c.NS)
+	}
+	fcn, _ := c.Doit.GetString(c.NS, "function")
+	list, err := c.Serverless().ListTriggers(context.TODO(), fcn)
```

Comment:
These commands use the DO API rather than the OpenWisk one, but they still depend on the `credentials.json` file to determine the connected namespace. It might be nice to guard them with a status check so that they provide a hint to connect if run without having done so instead of just `no such file or directory`.

    $ ./builds/doctl serverless triggers list 
    Error: open /home/asb/.config/doctl/sandbox/creds/ea841a9e/credentials.json: no such file or directory

Reply:
I did as suggested. 

In thinking about this issue, I suspect there are other cases where we should be guarding with a status check but are not.   The original idea was that that status check could be placed in utility functions that dispatch to the plugin rather that duplicating it in individual commands.   But, we have a gradually increasing number of commands that don't go through the plugin now and the whole area probably deserves a bit of refactoring.   I would like to defer that more general investigation to a follow-on, though, since I don't want to complicate this PR unduly.